simapp/abci.go

```diff
@@ -75,10 +185,63 @@ func (h *VoteExtensionHandler) VerifyVoteExtension() sdk.VerifyVoteExtensionHand
 		case !bytes.Equal(req.Hash, ve.Hash):
 			return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_REJECT}, nil
 
-		case len(ve.Data) != 1024:
-			return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_REJECT}, nil
 		}
-
 		return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_ACCEPT}, nil
 	}
 }
+
+// PreFinalizeBlock -> Extract VEs from previous block and persist in DB.
+func (app *SimApp) NewPreFinalizeBlockHookHandler() sdk.PreFinalizeBlockHook {
+	return func(ctx sdk.Context, req *abci.RequestFinalizeBlock) error {
+		if req.Height > 5 {
```

Comment:
why 5?

Reply:
The V.E. in the devnet I was testing were enabled from 5.